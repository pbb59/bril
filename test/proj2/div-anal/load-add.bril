# ARGS: diverge

main {
# init const idxs
init:
  zero: int = const 0;
  one: int = const 1;
  two: int = const 2;
  three: int = const 3;

# load some stuff into memory
  size: int = const 4;
  a: int = const 0;
  i: int = const 0;
sw_loop:
  ai: int = add a i;
  sw i ai;
  i: int = add i one;
  done: bool = eq i size;
  br done pack sw_loop;

pack:
  vec0: vector = vload zero;
  vec1: vector = vload zero;
add:
  vec2: vector = vadd vec0 vec1;
extract:
  a4: int = v2s vec2 zero;
  a5: int = v2s vec2 one;
  a6: int = v2s vec2 two;
  a7: int = v2s vec2 three;

# print results
print:
  print a4;
  print a5;
  print a6;
  print a7;
}
