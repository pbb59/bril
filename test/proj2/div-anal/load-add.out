main {
init:
  zero: int = const 0;
  one: int = const 1;
  two: int = const 2;
  three: int = const 3;
  size: int = const 4;
  a: int = const 0;
  i: int = const 0;
sw_loop:
  ai: int = add a i;
  sw i ai;
  i: int = add i one;
  done: bool = eq i size;
  br done pack sw_loop;
pack:
  vec0: vector = vload zero;
  vec1: vector = vload zero;
add:
  vec2: vector = vadd vec0 vec1;
extract:
  a4: int = v2s vec2 zero;
  a5: int = v2s vec2 one;
  a6: int = v2s vec2 two;
  a7: int = v2s vec2 three;
print:
  print a4;
  print a5;
  print a6;
  print a7;
}
init:
  in:  ∅
  out: ∅
sw_loop:
  in:  ∅
  out: ∅
pack:
  in:  ∅
  out: vec0, vec1
add:
  in:  vec0, vec1
  out: vec0, vec1, vec2
extract:
  in:  vec0, vec1, vec2
  out: a4, a5, a6, a7, vec0, vec1, vec2
print:
  in:  a4, a5, a6, a7, vec0, vec1, vec2
  out: a4, a5, a6, a7, vec0, vec1, vec2
0
2
4
6
0
2
4
6
